src/text_dispatcher.erl

```diff
@@ -0,0 +1,21 @@
+-module(text_dispatcher).
+-include_lib("eunit/include/eunit.hrl").
+
+%% API
+-export([dispatch/1]).
+
+process_count() -> 4.
+
+dispatch([]) -> empty_text;
+dispatch(Text) -> dispatch(Text, process_count(), 0, string:length(Text), []).
+dispatch(_, 0, _, _, PidList) -> lists:reverse(PidList);
+dispatch(Text, ProcessesLeft, SliceStart, TextLength, PidList) ->
+  SliceLength = TextLength div ProcessesLeft,
+  Pid = spawn(graph_processor, process, [string:slice(Text, SliceStart, SliceLength)]),
+  dispatch(Text, ProcessesLeft - 1, SliceStart + SliceLength, TextLength - SliceLength, [Pid|PidList]).
```

Comment:
Recurssion is nice, but we should use some more functional approach too. This could be done by preparing the list of tuples of `{SliceStart, SliceLenght}` first and then using `lists:map` to spawn the processes, like so:
```
lists:map(fun({SliceStart, SliceLenght}) -> spawn(graph_processor, process, [string:slice(Text, SliceStart, SliceLength)]) end, Slices)
```
where `Slices` is the list of prepared tuples (you could also prepare the list of sliced strings and use that list in map).

Reply:
I am not sure if I aggree. Firstly it is debatable what is more functional, using recursion, or doing some work, saving in some state, and then using `map` because `map` is functional xDD And if we are at least trying to go for speed, this is faster a little, since first process do not need to wait for the next one to get their piece of text sliced